### ISSUE TITLE
Cursor bug in vertical split

### ISSUE THREAD

**Issue by yalpul** [vim/vim#4245]
On vertical split mode with 2 windows, cursor doesn't show in the line immediately below when the command <kbd>o</kbd> is given or immediately above when <kbd>O</kbd> is given. This happens when one of the lines is spanning the whole horizontal space while at the same time line count is going from 99 to 100 (as a result of <kbd>o</kbd>). This happens when the window is either at the left side or right side in vertical split. It does not occur when there is only one window(no vertical split).

Detailed steps to reproduce the behavior:
1. Run `vim --clean` (or `gvim --clean`, etc.)
2. Enter `:set number` and `:set numberwidth=2`
3. Edit `file1`
4. Fill the file with 99 lines
5. Use `:vsp file2` to vertically split the window (content of `file2` is irrelevant)
6. In `file1`, fill one of the lines up to the point where the line is spanning the whole horizontal space(vertical split bar will be the bound)
7. In `file1` hit <kbd>o</kbd> or <kbd>O</kbd>. Cursor will be on the wrong place. When started typing, it writes correctly.

**Expected behavior**
Cursor being on the correct place, the line immediately below (or above).

**Screenshots**
![Screen Shot 2019-04-12 at 21 08 54](https://user-images.githubusercontent.com/16159263/56057391-6cce8800-5d67-11e9-9e05-864598b4a183.png)
*hit o (cursor is on `2`)*
![Screen Shot 2019-04-12 at 21 09 20](https://user-images.githubusercontent.com/16159263/56057447-8374df00-5d67-11e9-9c93-1feb9434c424.png)
*start typing*
![Screen Shot 2019-04-12 at 21 09 32](https://user-images.githubusercontent.com/16159263/56057469-8cfe4700-5d67-11e9-96ac-b8b591987845.png)

**Environment**
 - Vim version 8.0.1453(Ubuntu) and version 8.0.1283(macOS)
Edit: reproducable in version 8.1.1161
 - OS: Ubuntu 18.04 and macOS 10.14.4
 - Terminal: Apple Terminal App

**Additional context**
Basically, key points of the bug are: 

* Vertical split
* numberwidth=2
* line count going from 99 to 100
* one long line about to overflow to 2 lines

**Comment by Ruteri** [comment]
I have reproduced it in a slightly easier setup:
* happens without the split
* 9 lines are enough, I think what matters is the change of number of columns needed for line numbering
* does not happen when `numberwidth` is larger than number of columns required for numbering after the new line (e.g. for change 9->10 lines does not happen with `numberwidth` > 2, for 99->100 with `numberwidth` > 3), as the number of columns for numbering does not change
